preview/index.ts

```diff
@@ -0,0 +1,74 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { NonEmptyString } from 'io-ts-types';
+import { getEmailNewsletters, rowToNewsletter } from '../src/jobs/newsletters';
+import {
+	EmailNewsletter,
+	EmailNewsletterType,
+} from '../src/models/newsletters';
+import { parseStringifiedCSV } from './csv';
+
+const USE_LIVE_DATA = !!process.env.USE_LIVE_DATA;
+const PREVIEW_OUTPUT_FILE_PATH = './preview/preview.json';
+const PREVIEW_DATA_SOURCE_FILE_PATH = './preview/sampleData.csv';
+
+const logFeedback = (
+	versionNumber: string,
+	newsletters: EmailNewsletter[],
+): void => {
+	console.log({ versionNumber });
+	console.log('INDEX\t MATCH\t NAME');
+	newsletters.forEach((newsletter, index) => {
+		console.log(
+			index,
+			'\t',
+			EmailNewsletterType.is(newsletter),
+			'\t',
+			newsletter.name,
+		);
+	});
+};
+
+const getEmailNewslettersFromLocalCsv = async (): Promise<
+	EmailNewsletter[]
+> => {
+	const csvData = await readFileSync(
+		PREVIEW_DATA_SOURCE_FILE_PATH,
+	).toString();
+	const cellsInRows = parseStringifiedCSV(csvData);
+	const unverifiedNewsletters = cellsInRows.slice(1).map(rowToNewsletter);
+
+	let lastGroup = '_NO_GROUP_';
+	let lastTheme = '_NO_Theme_';
```

Comment:
looking at this again, it would be better to start with empty strings here so the values set on the first  `unverifiedNewsletters` will be empty if there is no value set on the first row of data, as would happen in the main `prepareRows` function.

Reply:
What is the reasoning behind handling "empty" `group` and `theme` by using a dummy string? If those fields are required in the model then we should probably require them here as well? As in, fail if they don't have a value. What do you think?

Reply:
It would be good to do another PR refactoring the `getEmailNewsletters` and `prepareRows` functions so the "fill in blank group and theme from row above" behaviour was in a separate function that coudl be used both on the data from the CSV and from the googlesheet rather than being duplicated.